vector/src/main/java/im/vector/app/core/extensions/Service.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.extensions
+
+import android.app.Notification
+import android.app.Service
+import android.content.pm.ServiceInfo
+import android.os.Build
+
+fun Service.startForegroundCompat(
+        id: Int,
+        notification: Notification,
+        provideForegroundServiceType: (() -> Int)? = null
```

Comment:
Using a lambda avoid warning about new API at the call site.

Reply:
Do you think there will be cases where we would need to declare the service type dynamically other than using the value declared in `AndroidManifest`?